app/services/x509/session_store.rb

```diff
@@ -0,0 +1,45 @@
+# Provides a way to access already-decrypted and cached PII from the redis
+# in an out-of-band fashion (using only the session UUID) instead of having access
+# to the user_session from Devise/Warden
+# Should only be used outside of a normal browser session (such as the OpenID Connect API)
+# See X509::Cacher for accessing PII inside of a normal browser session
+module X509
+  class SessionStore
+    attr_reader :session_uuid
+
+    def initialize(session_uuid)
+      @session_uuid = session_uuid
+    end
+
+    def ttl
+      uuid = session_uuid
+      session_store.instance_eval { redis.ttl(prefixed(uuid)) }
+    end
+
+    def load
+      session = session_store.send(:load_session_from_redis, session_uuid) || {}
+      X509::Attributes.new_from_json(session.dig('warden.user.user.session', :decrypted_x509))
+    end
+
+    # @api private
+    # Only used for convenience in tests
+    # @param [X509::Attributes] x509
+    def put(piv_cert_info, expiration = 5.minutes)
+      session_data = {
+        'warden.user.user.session' => {
+          decrypted_x509: piv_cert_info.to_h.to_json,
+        },
+      }
+
+      session_store.
+        send(:set_session, {}, session_uuid, session_data, expire_after: expiration.to_i)
+    end
+
+    private
+
+    def session_store
+      config = Rails.application.config
+      config.session_store.new({}, config.session_options)
+    end
+  end
+end
```

Comment:
This is pretty much the same as `Pii::SessionStore`. Is it worth it to extract this concept, or are we OK with this duplication?

Reply:
It's definitely the same. I copied and did a search/replace. Removed anything related to storing the data between sessions since we aren't cleared to do that yet. The reason for the copy is to use similar techniques as with PII but not tie this to how we are handling PII. Once all the dust settles from LOA3, I want to revisit.